internal/archive/zip/zip.go

```diff
@@ -28,6 +29,12 @@ func NewWriter(name string, matcher sauceignore.Matcher) (Writer, error) {
 	return w, nil
 }
 
+// New returns a new Writer that archives files to the specified io.Writer.
+func New(f io.Writer, matcher sauceignore.Matcher) (Writer, error) {
```

Comment:
`New` fails to make a distinction against`NewWriter`, since both are creating writers.
You can keep `New` as is, but then I'd recommend renaming `NewWriter` to `NewFileWriter`.

Reply:
This have been updated !